classes/ProductSale.php

```diff
@@ -235,28 +235,44 @@ public static function getBestSalesLight($idLang, $pageNumber = 0, $nbProducts =
      *
      * @param int $productId Product ID
      * @param int $qty Quantity
+     * @param int $idShop Shop ID
      *
      * @return bool Indicates whether the sale was successfully added
      */
-    public static function addProductSale($productId, $qty = 1)
+    public static function addProductSale($productId, $qty = 1, $idShop = null)
```

Comment:
@jolelievre, @kpodemski 
I now changed like suggested. 

But for these two functions `addProductSale` and `removeProductSale` working in `shop group` or `all shop` context is hurting the statistics. In both cases, we either add or remove one sale for each shop. This is duplicating sales or removing them duplicated.

We could argue, that it would be better do nothing, if `$isShop` is null, because in this case we can assess exactly the error: one sale has not be added or removed. When adding or removing on an array of shops, this array could be big, and the error introduced would be big too. (Suppose we have 20 shops, we would save 20 sales instead of 1).

Reply:
@lmeyer1, you are right. Unfortunately, I don't see any way to keep "BC Break-free" different way

We could:
- introduce a new method addProductSaleForShop
- deprecate or leave the old one
- change the use of addProductSale for addProductSaleForShop
- but still, there will be a problem with those who run addProductSale because `id_shop` will be `NULL`, and we won't get any information about it in methods getting sales

Maybe we should wait for PS 9.0 and change it then?